underlay/src/test/java/bio/terra/tanagra/filterbuilder/SurveyFilterBuilderTest.java

```diff
@@ -0,0 +1,510 @@
+package bio.terra.tanagra.filterbuilder;
+
+import static bio.terra.tanagra.utils.ProtobufUtils.serializeToJson;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.tanagra.api.filter.AttributeFilter;
+import bio.terra.tanagra.api.filter.BooleanAndOrFilter;
+import bio.terra.tanagra.api.filter.EntityFilter;
+import bio.terra.tanagra.api.filter.HierarchyHasAncestorFilter;
+import bio.terra.tanagra.api.filter.OccurrenceForPrimaryFilter;
+import bio.terra.tanagra.api.filter.PrimaryWithCriteriaFilter;
+import bio.terra.tanagra.api.shared.BinaryOperator;
+import bio.terra.tanagra.api.shared.Literal;
+import bio.terra.tanagra.api.shared.NaryOperator;
+import bio.terra.tanagra.filterbuilder.impl.core.SurveyFilterBuilder;
+import bio.terra.tanagra.proto.criteriaselector.DataRangeOuterClass.DataRange;
+import bio.terra.tanagra.proto.criteriaselector.KeyOuterClass.Key;
+import bio.terra.tanagra.proto.criteriaselector.ValueDataOuterClass.ValueData;
+import bio.terra.tanagra.proto.criteriaselector.ValueOuterClass.Value;
+import bio.terra.tanagra.proto.criteriaselector.configschema.CFSurvey;
+import bio.terra.tanagra.proto.criteriaselector.dataschema.DTSurvey;
+import bio.terra.tanagra.underlay.ConfigReader;
+import bio.terra.tanagra.underlay.Underlay;
+import bio.terra.tanagra.underlay.entitymodel.Hierarchy;
+import bio.terra.tanagra.underlay.entitymodel.entitygroup.CriteriaOccurrence;
+import bio.terra.tanagra.underlay.serialization.SZCorePlugin;
+import bio.terra.tanagra.underlay.serialization.SZService;
+import bio.terra.tanagra.underlay.serialization.SZUnderlay;
+import bio.terra.tanagra.underlay.uiplugin.CriteriaSelector;
+import bio.terra.tanagra.underlay.uiplugin.SelectionData;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class SurveyFilterBuilderTest {
```

Comment:
I think these tests should use actual survey entities instead of the same ones the entity group filter builder tests use. The filter builder tests are just testing that the right filter object is created, I don't think they're running any queries (you can confirm by running them in the `noCloudAccessRequiredTests` mode). So you could use the AoU config files here.

Reply:
These are removed now that the code is mostly shared. We can start adding survey specific tests when the functionality starts to diverge.